kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionFiller.java

```diff
@@ -0,0 +1,596 @@
+        } else if (wrappedExpression instanceof Relation) {
+            final Relation relationExpression = (Relation) wrappedExpression;
+            return new RelationProps(expressionName, dataType, columnsConvertForRelationProps(relationExpression), rowsConvertForRelationProps(relationExpression));
+        } else if (wrappedExpression instanceof List) {
+            final List listExpression = (List) wrappedExpression;
+            final Double width = listExpression.getComponentWidths().get(1);
+            return new ListProps(expressionName, dataType, itemsConvertForListProps(listExpression), width);
+        } else if (wrappedExpression instanceof Invocation) {
+            final Invocation invocationExpression = (Invocation) wrappedExpression;
+            final String invokedFunction = ((LiteralExpression) Optional.ofNullable(invocationExpression.getExpression()).orElse(new LiteralExpression())).getText().getValue();
+            final Double entryInfoWidth = invocationExpression.getComponentWidths().get(1);
+            final Double entryExpressionWidth = invocationExpression.getComponentWidths().get(2);
+            return new InvocationProps(expressionName, dataType, invokedFunction, bindingsConvertForInvocationProps(invocationExpression), entryInfoWidth, entryExpressionWidth);
+        } else if (wrappedExpression instanceof FunctionDefinition) {
+            final FunctionDefinition functionExpression = (FunctionDefinition) wrappedExpression;
+            final EntryInfo[] formalParameters = formalParametersConvertForFunctionProps(functionExpression);
+            final Double parametersWidth = functionExpression.getComponentWidths().get(1);
+            return specificFunctionPropsBasedOnFunctionKind(expressionName, dataType, functionExpression, formalParameters, parametersWidth);
+        } else if (wrappedExpression instanceof DecisionTable) {
+            final DecisionTable decisionTableExpression = (DecisionTable) wrappedExpression;
+            final String hitPolicy = decisionTableExpression.getHitPolicy() != null ? decisionTableExpression.getHitPolicy().value() : null;
+            final String aggregation = decisionTableExpression.getAggregation() != null ? decisionTableExpression.getAggregation().getCode() : "";
+            return new DecisionTableProps(expressionName, dataType, hitPolicy, aggregation,
+                                          annotationsConvertForDecisionTableProps(decisionTableExpression),
+                                          inputConvertForDecisionTableProps(decisionTableExpression),
+                                          outputConvertForDecisionTableProps(decisionTableExpression),
+                                          rulesConvertForDecisionTableProps(decisionTableExpression));
+        }
+        return new ExpressionProps(expressionName, dataType, null);
```

Comment:
Shouldn't we throw some exception about illegal state?

Reply:
I don't think. It is the case when an empty expression is translated to JS props.